src/main/scala/mesosphere/marathon/core/leadership/LeadershipModule.scala

```diff
@@ -0,0 +1,53 @@
+package mesosphere.marathon.core.leadership
+
+import akka.actor.{ ActorRef, ActorRefFactory, Props }
+import mesosphere.marathon.core.leadership.impl.{
+  LeadershipCoordinatorDelegate,
+  LeadershipCoordinatorActor,
+  WhenLeaderActor
+}
+
+/**
+  * This module provides a utility function for starting actors only when our instance is the current leader.
+  * This should be used for all normal top-level actors.
+  *
+  * In addition, it exports the coordinator which coordinates the activity performed when elected or stopped.
+  * The leadership election logic needs to call the appropriate methods for this module to work.
+  */
+class LeadershipModule(actorRefFactory: ActorRefFactory) {
+
+  private[this] var whenLeaderRefs = Set.empty[ActorRef]
+  private[this] var started: Boolean = false
+
+  /**
+    * Create a wrapper around an actor which should only be active if this instance of Marathon is the
+    * current leader instance. The wrapper starts the actor with the given props when appropriate and stops
+    * it when this instance loses Leadership. If the wrapper receives messages while not being the leader,
+    * it answers all messages with Status.Failure messages.
+    *
+    * @param props the props to create the actor
+    * @param name the name of the actor (the wrapping actor will be named like this)
+    * @param considerPreparedOnStart whether the actor is ready to receive messages as soon as it is started.
+    *                                if false, the actor expects a PreparationMessages.PrepareToStart message
+    *                                and is assumed to be ready when it replies with PreparationMessages.Prepared.
+    * @return
+    */
+  def startWhenLeader(props: => Props, name: String, considerPreparedOnStart: Boolean = true): ActorRef = {
```

Comment:
considerPreparedOnStart is never applied to the WhenLeaderActor - it is always considered prepared!
Perhaps get rid of the default parameter, so that everybody has to think about that explicitly.